cdap-formats/src/main/java/co/cask/cdap/format/DelimitedStringsRecordFormat.java

```diff
@@ -114,15 +134,118 @@ private boolean isStringArray(Schema schema) {
     return false;
   }
 
-  @Override
-  protected void configure(Map<String, String> settings) {
-    String charsetStr = settings.get(CHARSET);
-    if (charsetStr != null) {
-      this.charset = Charset.forName(charsetStr);
+  private static String parseBodyValue(String val, Schema fieldSchema) {
+    // if the body field is an empty string and the column is not a string type, interpret it as a null.
+    if (val != null && val.isEmpty() && (fieldSchema.getType() != Schema.Type.STRING)) {
+      return null;
     }
-    String delimiter = settings.get(DELIMITER);
-    if (delimiter != null) {
-      this.delimiter = delimiter;
+    return val;
+  }
+
+  private Map<String, Integer> parseMapping(String mappingString, Schema schema) {
+    Map<String, String> stringMapping = Splitter.on(",").withKeyValueSeparator(":").split(mappingString);
```

Comment:
think there is a way to make the splitter trim results, which would make this slightly more robust